org.eclipse.scout.sdk.s2i/src/main/kotlin/org/eclipse/scout/sdk/s2i/model/typescript/IdeaConstantValue.kt

```diff
@@ -38,21 +40,43 @@ open class IdeaConstantValue(val ideaModule: IdeaNodeModule, internal val elemen
     private val m_dataType = FinalValue<IDataType?>()
 
     fun unwrappedElement() = m_unwrappedElement.computeIfAbsentAndGet {
-        val unwrappedElement = unwrapTypeScriptAsExpression(element)
+        var unwrappedElement = unwrapJSParenthesizedExpression(element)
+        unwrappedElement = unwrapTypeScriptAsExpression(unwrappedElement)
+        unwrappedElement = unwrapTypeScriptExpressionWithTypeArguments(unwrappedElement)
         unwrapJSNewExpression(unwrappedElement)
     }
 
     override fun containingFile(): Optional<Path> = m_containingFile.computeIfAbsentAndGet { Optional.ofNullable(element?.containingFile?.virtualFile?.resolveLocalPath()) }
 
+    protected fun unwrapJSParenthesizedExpression(element: JSElement?) = if (element is JSParenthesizedExpression) element.innerExpression else element
```

Comment:
This only supports one level: (Class<TypeArg>). Nested Parenthesis is not supported. E.g. ((Class<TypeArg>)).
But maybe this is sufficient like this?

Reply:
I think this is sufficient as using parenthesis is no longer necessary in IJ 2023. Therefore, I think we do not need to spend extra time for something that is very unlikely to occur.

Reply:
I agree